### PR TITLE
RHDEVDOCS-2829 Explain how to "only deploy collector section"

### PR DESCRIPTION
**Pull request by rolfedh**
- For versions 4.5+
- https://issues.redhat.com/browse/RHDEVDOCS-2829
- Direct link to doc preview: https://deploy-preview-31141--osdocs.netlify.app/openshift-enterprise/latest/logging/config/cluster-logging-collector.html#cluster-logging-collecting-and-forwarding-logs-only_cluster-logging-collector
- Completed SME, QE, and peer reviews.
- **Ready for merge**.

### CODE REVIEW

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-collector.adoc
+
+[id="cluster-logging-collecting-and-forwarding-logs-only_{context}"]
+= Collecting and forwarding logs only
+
+As an administrator, you can collect and forward logs to an external log store and minimize resource consumption by eliminating other components.
```
Comment (syedriko) [inline modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:8]: CLF can forward logs to multiple destinations and they are not necessarily external to the OpenShift cluster.
Consumption of OpenShift cluster resources can be a reason to do it, if the destinations are off-cluster, but there can be other reasons as well, like secure storage of audit logs or use of an enterprise-wide logging store.

Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc]: Updated. Thanks!

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 2]
```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-collector.adoc
+
+[id="cluster-logging-collecting-and-forwarding-logs-only_{context}"]
+= Collecting and forwarding logs only
+
+As an administrator, if are forwarding logs to only a third-party log store and not to the default Elasticsearch log store, you can remove several unused components from your logging cluster.
```
Comment (syedriko) [inline modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:8]: >if are forwarding logs to only a third-party log store
if you are only forwarding logs to a third-party log store?

Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:8]: In casual conversation, I use "only" the way you suggest. However, the meaning of "only" can become more ambiguous the further you move it from the "exclusive object." For technical writing, it should be as close to the "exclusive object" as possible.
 
 For example, "if you are only forwarding logs to a third-party log store," can take on an alternative meaning, such as "only forwarding logs and not doing something else to them."

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+That is, you can preserve the Fluentd log `collection` component in the `ClusterLogging` custom resource (CR) and remove the Internal Elasticsearch `logStore`, Kibana `visualization`, and log `curation` components. Removing these components is optional but saves resources.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:10]: Thank you. Updated.

[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc]: Updated. Thank you.

**Comment by rolfedh** [comment]
@anpingli Please review [the proposed topic](https://deploy-preview-31141--osdocs.netlify.app/openshift-enterprise/latest/logging/config/cluster-logging-collector.html#cluster-logging-collecting-and-forwarding-logs-only_cluster-logging-collector). Thank you.

**Comment by anpingli** [comment]
@kabirbhartiRH  Could you look at this?

**Review by Preeticp — COMMENTED** [review]
Some minor nits, otherwise looks good to me.

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-collector.adoc
+
+[id="cluster-logging-collecting-and-forwarding-logs-only_{context}"]
+= Collecting and forwarding logs only
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:6]: We struggled to find a good title for this. Collecting and forwarding are always done, but almost never alone. I considered many permutations of this title. I'll give it another spin.

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+As an administrator, if you are forwarding logs to only a third-party log store and not to the default Elasticsearch log store, you can remove several unused components from your logging cluster.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:8]: Updated. Thanks.

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+That is, you can preserve the Fluentd log `collection` component in the `ClusterLogging` custom resource (CR) and remove the internal Elasticsearch `logStore`, Kibana `visualization`, and log `curation` components. Removing these components is optional but saves resources.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:10]: Updated. The second paragraph expands on the abstract in the first paragraph. "In other words" signals this fact to the reader, so they do not assume this is a new separate subject.

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+.Prerequisites
+
+* Verify that your log forwarder does not send log data to the default internal Elasticsearch cluster. Inspect the `ClusterLogForwarder` CR YAML file that you used to configure log forwarding. Verify that it **does not** have an `outputRefs` element that specifies `default`. For example:
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:14]: Thanks. Updated.

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
++
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc:15]: Updated. Thanks.

**Comment by rolfedh** [comment]
@kabirbhartiRH please also confirm that this is a supported configuration. Here's a link to the preview, for your convenience: https://deploy-preview-31141--osdocs.netlify.app/openshift-enterprise/latest/logging/config/cluster-logging-collector.html#cluster-logging-collecting-and-forwarding-logs-only_cluster-logging-collector

File: modules/cluster-logging-collecting-and-forwarding-logs-only.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (Preeticp) [reply modules/cluster-logging-collecting-and-forwarding-logs-only.adoc]: Though long the current title captures the gist of the module pretty well.

**Comment by kabirbhartiRH** [comment]
@rolfedh @anpingli Verified on 4.7 Cluster. I see the fluentd pods redeployed under openshift-logging ns.

LGTM!

**Comment by rolfedh** [comment]
Hi @preeti. I think this is ready to merge. Thanks!

**Pull request merged**